haas/api.py

```diff
@@ -602,20 +712,26 @@ def port_connect_nic(port, node, nic):
     db.commit()
 
 
-@rest_call('POST', '/port/<path:port>/detach_nic')
-def port_detach_nic(port):
+@rest_call('POST', '/switch/<switch>/port/<path:port>/detach_nic')
+def port_detach_nic(switch, port):
     """Detach a port from the nic it's attached to
 
     If the port does not exist, a NotFoundError will be raised.
 
     If the port is not connected to anything, a NotFoundError will be raised.
+
+    If the port is attached to a node which is not free, a BlockedError
+    will be raised.
     """
     db = model.Session()
 
-    port = _must_find(db, model.Port, port)
+    switch = _must_find(db, model.Switch, switch)
+    port = _must_find_n(db, switch, model.Port, port)
 
     if port.nic is None:
         raise NotFoundError(port.label + " not attached")
+    if port.nic.owner.project is not None:
```

Comment:
I can envision scenarios where one might want to change the switch port of an allocated node. For instance, if a port/cable is dead, it could be moved over to another open port.